File: 이승연/프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        Map<String, Integer> map = new HashMap<>();
        
        for (int i = 0; i < players.length; i++) {
            map.put(players[i], i);
        }
        
        for (int i = 0; i < callings.length; i++) {
            String backPlayer = callings[i];
            String frontPlayer = players[map.get(backPlayer) - 1];
            
            int frontIndex = map.get(frontPlayer) + 1;
            int backIndex = map.get(backPlayer) - 1;
            
            map.put(frontPlayer, frontIndex);
            map.put(backPlayer, backIndex);
            
            players[frontIndex] = frontPlayer;
            players[backIndex] = backPlayer;
        }
        
        return players;
    }
}

File: 이승연/프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    const data = {};
    
    for (let i = 0; i < players.length; i++) {
        data[players[i]] = i;
    }
    
    for (let i = 0; i < callings.length; i++) {
        const backPlayer = callings[i];
        const frontPlayer = players[data[backPlayer] - 1];
        
        const backIndex = data[backPlayer] - 1;
        const frontIndex = data[frontPlayer] + 1;
        
        data[backPlayer] = backIndex;
        data[frontPlayer] = frontIndex;
        
        players[backIndex] = backPlayer;
        players[frontIndex] = frontPlayer;
    }
    
    return players;
}